### PR TITLE
Comments

### PR DESCRIPTION
**Pull request by hariso**

### CODE REVIEW

**Comment by WeirdMagician** [comment]
Thanks @hariso @maha-hajja for the comments. Let me merge and start working on to dos.

**Comment by hariso** [comment]
@WeirdMagician I can also suggest you taking a loot at the comments and letting us know if you have any additional questions. Then we can turn the comments into issues, so we can more easily split the work (track it and review it).

**Comment by WeirdMagician** [comment]
Thanks for providing the review, Fixed few of them.
Currently below are the consolidated issues will update/add if any during our conversions.

1. set up workflows for this repository (build, test, lint etc.)
2. snapshot iterator integration testcases
3. cdc iterator integration testcases
4. source testcases
5. reusable packages with S3 and GCS 
6. position to be serialize as JSON

File: source/position/position.go [hunk 1]
```diff
@@ -28,6 +28,7 @@ type Position struct {
 	Type      Type
 }
 
+// maybe it would be simpler if we just serialize it as JSON?
```
[the remark this answers is not in the document]
Comment (hariso) [reply source/position/position.go:31]: No constraints to my knowledge.

**Comment by maha-hajja** [comment]
@WeirdMagician we noticed that we open connection to S3 in the "Configure" method instead of "Open" method in the S3 connector.. and fixed it https://github.com/ConduitIO/conduit-connector-s3/pull/43

File: source/iterator/cdc_iterator.go [hunk 2]
```diff
@@ -1,3 +1,7 @@
+// todo: is this the best way to get CDC records from GCS?
```
[the remark this answers is not in the document]
Comment (hariso) [reply source/iterator/cdc_iterator.go:1]: If you could investigate both approaches (current, and with pub/sub), and compare them, that would be great.

[the remark this answers is not in the document]
Comment (WeirdMagician) [reply source/iterator/cdc_iterator.go:1]: Sure let me create a issue for tracking the progress

**Comment by WeirdMagician** [comment]
@hariso Updates on the previously consolidated issues

1.     set up workflows for this repository (build, test, lint etc.) - current status is closed but can reopen if there are any comments. - https://github.com/WeirdMagician/conduit-connector-google-cloudstorage/issues/2

Joined the below 3 items in to a single issue - current status is open - https://github.com/WeirdMagician/conduit-connector-google-cloudstorage/issues/5
2.     snapshot iterator integration testcases
3.     cdc iterator integration testcases
4.     source testcases


5.     reusable packages with S3 and GCS - As we have discussed earlier that we can ignore it now
6.     position to be serialize as JSON - https://github.com/WeirdMagician/conduit-connector-google-cloudstorage/issues/3 - current status is closed but can reopen if there are any comments.
7. Explore Pub/sub for cdc - https://github.com/WeirdMagician/conduit-connector-google-cloudstorage/issues/11 - current status is open

**Comment by maha-hajja** [comment]
just making sure that u got this comment https://github.com/WeirdMagician/conduit-connector-google-cloudstorage/pull/1#issuecomment-1118462002

File: config/config.go [hunk 3]
```diff
@@ -1,3 +1,5 @@
+// maybe use a single package?
+// set up workflows (build, test, lint etc.)
```
[the remark this answers is not in the document]
Comment (hariso) [reply config/config.go:2]: :+1:

File: source/iterator/cdc_iterator.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (hariso) [reply source/iterator/cdc_iterator.go:1]: Discussed during the demo today, so I'll mark as resolved.

File: source/iterator/snapshot_iterator.go [hunk 4]
```diff
@@ -11,6 +11,10 @@ import (
 	sdk "github.com/conduitio/conduit-connector-sdk"
 )
 
+// would it be possible to actually reuse the S3 connector?
+// i.e. import as a library and use it as such?
+// GCS and S3 are built pretty much on same concepts.
```
[the remark this answers is not in the document]
Comment (hariso) [reply source/iterator/snapshot_iterator.go:16]: We discussed this and copying code is fine for now.

File: source/source_test.go [hunk 5]
```diff
@@ -19,6 +20,7 @@ func TestConfigureSource_FailsWhenConfigInvalid(t *testing.T) {
 	con := Source{}
 	err := con.Configure(context.Background(), map[string]string{"foobar": "foobar"})
 
+	// same comments as above
```
Comment (hariso) [inline source/source_test.go:23]: @WeirdMagician when you get to it, please check the above comments, just so we are sure we got everything covered.

File: config/config.go [hunk 6]
```diff
@@ -9,6 +11,7 @@ import (
 )
 
 const (
+	// maybe shorten?
```
[the remark this answers is not in the document]
Comment (WeirdMagician) [reply config/config.go:14]: Actually it is completed and added to main branch when this PR is created. we didn't tracked it separately and thanks for reminding.
I have changed the identifiers to
	GoogleCloudStorageBucket
	GoogleCloudServiceAccountKey

File: source/source_test.go (hunk 5 above)
Comment (WeirdMagician) [reply source/source_test.go:23]: Actually it is completed and added to main branch when this PR is created. we didn't tracked it separately and thanks for reminding.
I have used every where errors.is whenever an expected err is already available if not using the string comparing.

File: config/config.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (hariso) [reply config/config.go:14]: Great!

**Pull request closed without merging**